web/src/features/fireWeather/pages/MoreCastPage.tsx

```diff
@@ -68,7 +70,21 @@ const MoreCastPage = () => {
 
   const shouldInitiallyShowSidePanel = codesFromQuery.length > 0
   const [showSidePanel, setShowSidePanel] = useState(shouldInitiallyShowSidePanel)
-  const openSidePanel = () => setShowSidePanel(true)
+  const [sidePanelWidth, setSidePanelWidth] = useState(
+    shouldInitiallyShowSidePanel ? PARTIAL_WIDTH : 0
+  )
+
+  const [mapCenter, setMapCenter] = useState(CENTER_OF_BC)
+  const expandSidePanel = () => setSidePanelWidth(FULL_WIDTH)
+  const collapseSidePanel = () => setSidePanelWidth(PARTIAL_WIDTH)
+
+  const setNewMapCenter = (newMapCenter: number[]) => {
+    setMapCenter(newMapCenter)
```

Comment:
Maybe some comments around why this is needed?

Reply:
Added clarifying comment